### PR TITLE
Add command line to runs tests repeatedly

### PR DESCRIPTION
**Pull request by mjp41**
Adds a 
```
-r=[N]
```
to run the tests N times.

The PR also fixes the `-procs=` and `-j` option, which were ignored.

### CODE REVIEW

File: bench.sh [hunk 1]
```diff
@@ -407,7 +411,10 @@ benchfill="           "
 function run_test_env_cmd { # <test name> <allocator name> <environment args> <command>
   echo
   echo "run $1 $2: $3 $4"
-  # cat $benchres
+  # clear temporary output
+  if [ -f $benchres.line ]; then
+    rm $benchres.line
```
Comment (jvoisin) [inline bench.sh:416]: please add quotes about `$benchres.line` and prefix it with `./`, just to be on the safe side.

File: bench.sh [hunk 2]
```diff
@@ -445,41 +452,42 @@ function run_test_env_cmd { # <test name> <allocator name> <environment args> <c
...
       rtime=`echo "scale=3; (10000000000 / $ops)" | bc`
       echo "$1,$2: iterations: ${ops}, relative time: ${rtime}s"
-      sed -E -i.bak "s/($1  *$2  *)[^ ]*/\10:$rtime/" $benchres;;
+      sed -E -i.bak "s/($1  *$2  *)[^ ]*/\10:$rtime/" $benchres.line;;
     spec-*)
       popd;;
   esac
-  tail -n1 $benchres
+  cat $benchres.line
+  cat $benchres.line >> $benchres
```
Comment (jvoisin) [inline bench.sh:490]: You can use `tee` here instead of `cat` twice.

File: bench.sh (hunk 1 above)
Comment (mjp41) [reply bench.sh:416]: So `$benchres` is a full path, so prepending with `./` does not work.  Can I ask what you want to be safe with respect to?

Comment (jvoisin) [reply bench.sh:416]: If it's a full path, then no need for `./`, but quotes are still needed, should `$benchres.line` contain a space character.

File: bench.sh (hunk 2 above)
Comment (mjp41) [reply bench.sh]: Used `tee -a` to continue to append.

File: bench.sh (hunk 1 above)
Comment (mjp41) [reply bench.sh]: Fixed this. I think there are multiple places where this script will fail if there are spaces in the directory where it is stored.

Comment (jvoisin) [reply bench.sh]: We should fix them all, before someone reports a "mimalloc-bench eat all my files :'("

Comment (mjp41) [reply bench.sh]: I fixed the ones for `rm`.  I'll look at fixing the rest.

**Comment by daanx** [comment]
Apologies for letting this linger! Will merge now

**Pull request merged**